plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/OpenShiftResourceUIModel.java

```diff
@@ -19,30 +21,41 @@
  * Display model for OpenShift resources to allow
  * overriding equals and hash
  * @author jeff.cantrill
+ * @author Jeff Maury
  *
  */
 public class OpenShiftResourceUIModel extends ObservableUIPojo implements IResourceUIModel{
 
-	private final IResource resource;
-	private final String tostring;
+	private IResource resource;
+	private String tostring;
 	private final Object parent;
+	private AtomicBoolean deleting = new AtomicBoolean(false);
+
 
-	public OpenShiftResourceUIModel(IResource resource, IResourcesUIModel parent){
+	public OpenShiftResourceUIModel(IResource resource, Object parent){
```

Comment:
The rationale for the refactoring was that every UI element owns the mapped Openshift resource thus kept as the root of the hierarchy. Deployement kept a reference to the service where the mapped resource could be used. Please not that as we want to keep a flag for storing the deleting state, then we need a common class to store this state and so the the resource